lib/vf_settokenvalue.cpp

```diff
@@ -550,17 +551,18 @@ namespace ValueFlow
                         setTokenValue(parent, std::move(result), settings);
                     } else if (Token::Match(parent, "%op%")) {
                         if (Token::Match(parent, "%comp%")) {
-                            if (!result.isFloatValue() && !value1.isIntValue() && !value2.isIntValue())
+                            if (!isFloat && !value1.isIntValue() && !value2.isIntValue())
                                 continue;
                         } else {
                             if (value1.isTokValue() || value2.isTokValue())
                                 break;
                         }
                         bool error = false;
+                        auto val = calculate(parent->str(), floatValue1, floatValue2, &error);
```

Comment:
You can't calculate the integral values using floating-point. If both value1 and value2 are integrals then you should use the intvalue to calculate.

Reply:
But it works... 😃

> Actually, looking at the test, we just need to run calculate as both float if one operand is float and convert to the result type int or float at the end.

Reading this again I see I got it wrong. So re-introduce the previous calculation code but also keep the assignment code I added. Will do later.

Reply:
Adjusted it and it still passed locally. Duplicated some code for now.